docs/program/extend/modular-resources/_index.md

```diff
@@ -1,150 +1,528 @@
+        pass
 
-{{% alert title="Modules vs modular resources" color="tip" %}}
-Modules are binary executables that can be managed by the RDK through [module configuration](#add-a-module-to-your-robot-configuration).
+    # set_velocity: unimplemented
+    async def set_velocity( self, linear: Vector3, angular: Vector3, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+        pass
+
+    # set_power: sets the linear and angular velocity of the left and right motors on the base
+    async def set_power(self, linear: Vector3, angular: Vector3, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+
+        # stop the base if absolute value of linear and angular velocity is less than .01 
+        if abs(linear.y) < 0.01 and abs(angular.z) < 0.01:
+            return self.stop(extra=extra, timeout=timeout)
+
+        # use linear and angular velocity to calculate percentage of max power to pass to SetPower for left & right motors 
+        sum = abs(linear.y) + abs(angular.z)
+
+        self.left.set_power(power=((linear.y - angular.z) / sum), extra=extra, timeout=timeout)
+        self.right.set_power(power=((linear.y + angular.z) / sum), extra=extra, timeout=timeout)
+
+    # stop: stops the base from moving by stopping both motors
+    async def stop(self, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+        self.left.stop(extra=extra, timeout=timeout)
+        self.right.stop(extra=extra, timeout=timeout)
+
+    # is_moving: checks if either motor on the base is moving with motors' is_powered
+    async def is_moving(self, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs) -> bool:
+        return self.left.is_powered(extra=extra, timeout=timeout)[0] or self.right.is_moving(extra=extra, timeout=timeout)[0]
+```
+
+<file>init.py</file>
+
+``` python {class="line-numbers linkable-line-numbers"}
+"""
+This file registers the MyBase model with the Viam Registry.
+"""
+
+from viam.components.motor import *
+from viam.components.base import Base
+from viam.resource.registry import Registry, ResourceCreatorRegistration
+
+from .my_base import MyBase
+
+Registry.register_resource_creator(Base.SUBTYPE, MyBase.MODEL, ResourceCreatorRegistration(MyBase.new, MyBase.validate_config))
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{% /expand%}}
+
+{{% alert title="Caution" color="caution" %}}
+
+You must define all functions belonging to a built-in resource type if defining a new model. Otherwise, the class won’t instantiate.
+
+- If you are using the Python SDK, raise an `NotImplementedError()` in the body of functions you do not want to implement or put `pass`.
+- If you are using the Go SDK, return `errUnimplemented` or leave the body of functions you do not want to implement empty.
```

Comment:
Go function bodies can't be empty. And functions that have a return signature must follow that signature. MOST can probably return just errUnimplemented, but others may have to create and return correctly typed (but empty) values.

Reply:
Ok addressed, thanks!